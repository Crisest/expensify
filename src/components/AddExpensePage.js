import React from 'react'

const AddExpensePage = () => (
    <div>
        This is from my Add expense
    </div>
)

export default AddExpensePage