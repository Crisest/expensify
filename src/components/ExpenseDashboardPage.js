import React from 'react'

const ExpenseDashboardPage = () => (
    <div>
        This is from my dashboardcomponent
    </div>
)

export default ExpenseDashboardPage
