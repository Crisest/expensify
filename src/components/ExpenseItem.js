import React from 'react';
import { Link } from 'react-router-dom'

const ExpenseItem = ({ description, amount, createdAt, id, dispatch}) => (
    <div>
        <h3><Link to={`edit/${id}`}>{description}</Link></h3>
        <p>{amount} - {createdAt}</p>
    </div>
)

export default ExpenseItem