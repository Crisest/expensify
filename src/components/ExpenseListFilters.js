import React from 'react';
import { connect } from 'react-redux'
import { DateRangePicker } from 'react-dates'
import { setFilterText, sortByDate, sortByAmount, setStartDate, setEndDate } from '../actions/filters'

class ExpenseListFilters extends React.Component {
    state = { 
        calendarFocused: null
    }

    onDateChange = ({startDate, endDate}) => {
        this.props.dispatch(setStartDate(startDate))
        this.props.dispatch(setEndDate(endDate))
    }
    onFocusChange = (calendarFocused) => {
        this.setState({calendarFocused})
    }
    render(){
        return (
            <div className="container">
            <div className="input-group">
                <div className="input-group__item">
                    <input 
                    type="text" 
                    className="text-input" 
                    value={this.props.filters.text} 
                    onChange={(e) => {
                        this.props.dispatch(setFilterText(e.target.value))
                    }}
                    placeholder="Search Expenses"
                    />
                 </div>
                <div className="input-group__item">
                    <select 
                    value={this.props.filters.sortBy}
                    onChange={(e) => {
                        if(e.target.value === 'date'){
                            this.props.dispatch(sortByDate())
                        }
                        else if( e.target.value === 'amount'){
                            this.props.dispatch(sortByAmount())
                        }
                    }}
                    className="select"
                    >
                        <option value="date">Date</option>
                        <option value="amount">Amount</option>
                    </select>
                </div>
                <div className="input-group__item">
                    <DateRangePicker
                    startDate={this.props.filters.startDate}
                    endDate={this.props.filters.endDate}
                    onDatesChange={this.onDateChange}
                    focusedInput={this.state.calendarFocused}
                    onFocusChange={this.onFocusChange}
                    numberOfMonths={1}
                    isOutsideRange={() => false}
                    showClearDates={true}
                    startDateId="UniqueID"
                    endDateId="SDS"
                    />
                </div>
            </div>
            </div>
        )
    }
}



const ConnectedExpenseListFilters = (state) => {
    return {
        filters: state.filters
    }
}

export default connect(ConnectedExpenseListFilters)(ExpenseListFilters)