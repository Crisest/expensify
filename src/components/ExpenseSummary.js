import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom'
import selectExpenseTotal from '../selectors/selectExpenseTotal'
import selectExpenses from '../selectors/expenses'
import numeral from 'numeral'

export const ExpenseSummary = (props) => {
    const expenseWord = props.expenseCount === 1 ? 'expense' : 'expenses'
    return (
        <div className="page-header">
            <div className="container">
                <h1 className="page-header__title"> Viewing <span>{props.expenseCount}</span> {expenseWord} totalling <span>{numeral(props.expenseTotal / 100).format('$0,0.00')}</span></h1>
                <div className="page-header__actions">
                    <Link className="button" to="/create">Add Expense</Link>
                </div>
            </div>          
        </div>
    )
}

const mapStatetoProps = (state) => ({
    expenseTotal: selectExpenseTotal(selectExpenses(state.expenses, state.filters)),
    expenseCount: selectExpenses(state.expenses, state.filters).length
})

export default connect(mapStatetoProps)(ExpenseSummary)