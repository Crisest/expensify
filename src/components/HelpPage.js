import React from 'react'

const HelpPage = () => (
    <div>
        This is from the help Page
    </div>
)

export default HelpPage
