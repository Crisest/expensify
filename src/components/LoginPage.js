import React from 'react';

const LoginPage = () => (
    <div>
        <form>
            <label>Username</label>
            <input type="text" />
            <label>Password</label>
            <input type="text" />
            <button>Login</button>
        </form>
    </div>
)