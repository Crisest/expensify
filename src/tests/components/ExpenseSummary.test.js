import React from 'react';
import { shallow } from 'enzyme'
import { ExpenseSummary } from '../../components/ExpenseSummary';

test('should correctly render ExpenseSummary with 1 expense', () => {
    const wrapper = shallow(<ExpenseSummary expenseCount={1} expenseTotal={123}  />)
    expect(wrapper).toMatchSnapshot()
})

test('should correctly render ExpenseSummary with multiple expenses', () => {
    const wrapper = shallow(<ExpenseSummary expenseCount={22} expenseTotal={123456}  />)
    expect(wrapper).toMatchSnapshot()
})