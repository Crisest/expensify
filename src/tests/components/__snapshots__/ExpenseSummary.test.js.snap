// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpenseSummary with 1 expense 1`] = `
<div
  className="page-header"
>
  <div
    className="container"
  >
    <h1
      className="page-header__title"
    >
       Viewing 
      <span>
        1
      </span>
       
      expense
       totalling 
      <span>
        $1.23
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should correctly render ExpenseSummary with multiple expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="container"
  >
    <h1
      className="page-header__title"
    >
       Viewing 
      <span>
        22
      </span>
       
      expenses
       totalling 
      <span>
        $1,234.56
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
